File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    let visited = Array(n).fill(false);
    for(let i = 0; i < computers.length; i++){
        if(visited[i] === false) {
            dfs(i, n, visited, computers);
            answer += 1;
        }
    }
    return answer;
}

function dfs(i, n, visited, computers) {
    visited[i] = true;
    for(let j = 0; j < n; j++){
        if(i !== j && computers[i][j] === 1) {
            if(!visited[j]){
                dfs(j, n, visited, computers)
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False for _ in range(n)]
    for i in range(n):
        if visited[i] == False:
            dfs(i, visited, n, computers)
            answer += 1
    return answer

def dfs(i, visited, n, computers):
    visited[i] = True
    for j in range(n):
        if i != j and computers[i][j]:
            if not visited[j]:
                dfs(j, visited, n, computers)
    